ocaml/xapi/nm.ml

```diff
@@ -618,6 +596,36 @@ let bring_pif_up ~__context ?(management_interface = false) (pif : API.ref_PIF)
                     in
                     (conf, gateway)
               in
+              let static =
+                match address_type with
+                | `IPv4 -> rc.API.pIF_ip_configuration_mode = `Static
+                | `IPv6 -> rc.API.pIF_ipv6_configuration_mode = `Static
+              in
+              let dns =
+                if static
+                then (
+                  if rc.API.pIF_DNS <> "" then (
+                    let nameservers =
+                      List.map Unix.inet_addr_of_string
+                        (String.split ',' rc.API.pIF_DNS)
+                    in
+                    let domains =
+                      if List.mem_assoc "domain" rc.API.pIF_other_config
+                      then (
+                        let domains =
+                          List.assoc "domain" rc.API.pIF_other_config
+                        in
+                        try String.split ',' domains
+                        with _ ->
+                          warn "Invalid DNS search domains: %s" domains ;
+                          []
+                      ) else []
```

Comment:
Consider using `match List.assoc_opt with None -> [] | Some domains -> ...`

Reply:
instead of using both `mem_assoc` and `assoc` right?

Reply:
That's correct, that would mean traversing the list once instead of twice.